From: Luca Beltrame <lbeltrame@kde.org>
Date: Mon, 01 Oct 2012 18:47:56 +0000
Subject: Remove duplicated QVector<int> definition, since it's in PyQt now.
X-Git-Url: http://quickgit.kde.org/?p=pykde4.git&amp;a=commitdiff&amp;h=017822bd0dfc83fe9a7a483ecc33f4aab839a3c6
---
Remove duplicated QVector<int> definition, since it's in PyQt now.
Simon, if you have time, please review if everything is OK.

CCMAIL: simon@simonzone.com
---


--- a/sip/kdecore/typedefs.sip
+++ b/sip/kdecore/typedefs.sip
@@ -951,78 +951,4 @@
 %End
 };
 
-%MappedType QVector<int>
-{
-%TypeHeaderCode
-#include <qvector.h>
-%End
-
-%ConvertFromTypeCode
-    // Create the list.
-    PyObject *l;
-
-    if ((l = PyList_New(sipCpp->size())) == NULL)
-        return NULL;
-
-    // Set the list elements.
-    for (int i = 0; i < sipCpp->size(); ++i)
-    {
-        int t = (sipCpp->at(i));
-
-#if PY_MAJOR_VERSION >= 3        
-        PyObject *tobj = PyLong_FromLong(t);
-#else
-        PyObject *tobj = PyInt_FromLong(t);
-#endif
-
-        PyList_SET_ITEM(l, i, tobj);
-    }
-
-    return l;
-%End
-
-%ConvertToTypeCode
-    // Check the type if that is all that is required.
-    if (sipIsErr == NULL)
-    {
-        if (!PyList_Check(sipPy))
-            return 0;
-
-        for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i) {
-            PyObject *tobj = PyList_GET_ITEM(sipPy, i);
-#if PY_MAJOR_VERSION >= 3
-            if (!PyNumber_Check(tobj))
-#else
-            if (!PyInt_Check(tobj))
-#endif
-                return 0;
-        }
-        return 1;
-    }
-
-    QVector<int> *qv = new QVector<int>;
- 
-    for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i)
-    {
-        PyObject *tobj = PyList_GET_ITEM(sipPy, i);
- #if PY_MAJOR_VERSION >= 3
-        int t = PyLong_AsLong (tobj);
-#else
-        int t = PyInt_AS_LONG (tobj);
-#endif
-
-        if (*sipIsErr)
-        {
-            delete qv;
-            return 0;
-        }
-
-        qv->append(t);
-    }
- 
-    *sipCppPtr = qv;
- 
-    return sipGetState(sipTransferObj);
-%End
-};
-
+

